src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -29,15 +35,20 @@
   private final String INVALID_INPUT_ERROR = "Video is private or does not exist.";
   private final String COMMENTS_FAILED_ERROR = "Comments could not be retrieved.";
   private final String NLP_API_ERROR = "Language service client failed.";
-  private final String NO_DATA_ERROR = "No comments or captions available to analyze.";
-  private final String READER_ERROR = "Reading video ID failed.";
 
   public SentimentServlet() throws IOException, GeneralSecurityException {
     languageService = LanguageServiceClient.create();
     captionService = new Caption();
     commentService = new CommentService();
   }
 
+  /**
+   * Retrieves the comments and captions of a YouTube video, given its video ID, and calculates the
```

Comment:
this method seems to have a lot of responsibilities now. I suggest breaking some of those responsibilities (like calling the sentiment API) out into separate classes. That should make each component easier to understand and test